shell/src/main/java/alluxio/cli/fs/command/CpCommand.java

```diff
@@ -91,20 +154,36 @@ public int run(CommandLine cl) throws AlluxioException, IOException {
     AlluxioURI dstPath = new AlluxioURI(args[1]);
     if ((dstPath.getScheme() == null || isAlluxio(dstPath.getScheme()))
         && isFile(srcPath.getScheme())) {
-      List<File> srcFiles = FileSystemShellUtils.getFiles(srcPath.getPath());
-      if (srcFiles.size() == 0) {
-        throw new IOException(ExceptionMessage.PATH_DOES_NOT_EXIST.getMessage(srcPath));
-      }
+      List<AlluxioURI> srcPaths = new ArrayList<>();
       if (srcPath.containsWildcard()) {
-        List<AlluxioURI> srcPaths = new ArrayList<>();
+        List<File> srcFiles = FileSystemShellUtils.getFiles(srcPath.getPath());
+        if (srcFiles.size() == 0) {
+          throw new IOException(ExceptionMessage.PATH_DOES_NOT_EXIST.getMessage(srcPath));
+        }
         for (File srcFile : srcFiles) {
           srcPaths.add(
               new AlluxioURI(srcPath.getScheme(), srcPath.getAuthority(), srcFile.getPath()));
         }
-        copyFromLocalWildcard(srcPaths, dstPath);
       } else {
-        copyFromLocal(srcPath, dstPath);
+        File src = new File(srcPath.getPath());
+        if (src.isDirectory()) {
```

Comment:
could we use `FileSystemShellUtils.getFiles()` here as well?

Reply:
```
  /**
   * Gets the files (on the local filesystem) that match the given input path.
   * If the path is a regular path, the returned list only contains the corresponding file;
   * Else if the path contains wildcards, the returned list contains all the matched Files.
   *
   * @param inputPath The input file path (could contain wildcards)
   * @return a list of files that matches inputPath
   */
  public static List<File> getFiles(String inputPath) {```

it will only recursively traverse a directory if the path contains wildcard.